### ISSUE TITLE
Logitech F310/F710 do not work together with Logitech Rumblepad 2 and/or Dual Action

### ISSUE THREAD

**Issue by s7ygian** [OpenEmu/OpenEmu#2326]
I have realized strange behavior regarding some controller combinations.

Everything works as expected after plugging in:
- Logitech F310 only
- Logitech F710 only
- Logitech Rumblepad 2 only
- Logitech Dual Action only
- Logitech Rumblepad 2 together with Dual Action

However control messed up after plugging in:
- Logitech F310/F710 together with Rumblepad 2
- Logitech F310/F710 together with Dual Action
- Logitech F310/F710 together with Rumblepad 2 and Dual Action

In last 3 cases playing is not possible anymore because at least one connected gamepad does not respond during control mapping (tested for SNES, Genesis and N64).

Logitech F310 and F710 have been switched to DirectInput mode.

Any help?

Used software versions:

OSX v10.11.5
OpenEmu v2.0.2

**Comment by hoppena** [comment]
The same happens to me: I have a F710 and a Rumblepad 2: the controller assigned to player 1 works perfectly, while the one assigned to player 2 recognizes only the buttons and not the directions buttons or pad.
Any solution?
Thanks

OSX 10.11.3
OpenEmu 2.0.1

**Comment by AeonPhoenix** [comment]
I have the EXACT same issue. I have two older Rumblepad 2's, and a newer F310 and they won't play nice together. I loaded up enjoyable as a keyboard mapping software program, and that recognizes all 3 at the same time, just OpenEmu won't. I've also tried mapping them to keys on the keyboard but it still won't recognize one of the controllers, usually the F310.

OSX 10.11.6
OpenEmu 2.0.3

**Comment by clobber** [comment]
@s7ygian Thank you for the original detailed report! It took time to acquire these devices (Logitech F310 and Logitech Dual Action) to test with, but thanks to having set up donations we can finally check out issues with gamepads if we need to.

So, despite all of these devices reporting as "Logitech Dual Action" when set to DirectInput mode and having matching vendor/product ID's, they do not all have the exact HID profiles. With the newer F310/F710 models, they removed a HID collection grouping from the profile so that all of the cookie elements for the buttons are off by one. This makes the devices differ enough to throw off our input automap system which ends up not working with both a F310/F710 and Rumblepad 2/Dual Action plugged in.

As you can see with the profile screenshots below, the cookie elements for the buttons do not match up:

**Logitech Dual Action profile**
![logitech dual action](https://cloud.githubusercontent.com/assets/1303132/17715291/1127c7fe-63c9-11e6-9cff-eb08ae159878.png)

**Logitech F310 profile**
![logitech f310](https://cloud.githubusercontent.com/assets/1303132/17715324/2e60971a-63c9-11e6-971d-9ac82511a1dc.png)

It will take some time to figure out a way to cleanly differentiate these devices and still be able to automap controls for them, so for now the workaround is to either use only F310/F710 devices together or Rumblepad 2/Dual Action devices together. Alternatively, you may be able to delete the entire "OEControllerLogitechDualAction" dictionary from the Controller-Database.plist inside the OpenEmu application bundle which should allow you to use all devices at once, but then you will have to remap controls.

**Comment by seppiz** [comment]
Hi, I have the same problem with the two controllers Logitech DualAction/F310. Deleting the "OEControllerLogitechDualAction" dictionary from the Controller-Database.plist did not help. When I try to map the controls, first is fine, but for the second controller it is not recognized any key. For the buttons I can see exception like
```
Controller type 'OEControllerLogitechDualAction' does not recognize the event
'<OEHIDEvent 0x618000094730 pad=0x6000000a5760 type=Button number=3 state=On
'Button 3' cookie=6>', when attempting to set the key with name: 'OESNESButtonA'.
```
Any workaround idea on this is welcome (maybe use the locationId for unique hashes in controlValueIdentifierForType?), thanks.

**Comment by zhmzhmzhm** [comment]
Hey guys, 

any updates on that matter? 

I have two controllers - f310 and f710, the first one works fine and is seen by openemu as logitec dual action, the second one as logitech cordless rumble pad 2. 
I cannot assign/map action buttons on F710 (x, a, b, y) all other buttons ar functioning properly. 

I checked with usb overdrive, the system recognizes inputs from all the buttons, so it must be an issue with openemu. 

In case any additional info is needed, I am willing to help. 

thanks!

**Comment by oneklc** [comment]
Can confirm this issue is still present.  Any updates?
Can also confirm that a playstation controller with either the f310 or the f710 will work simultaneously but the logitech controllers still don't play nice with each other...

**Comment by tholdawa** [comment]
I'm also seeing this issue when using the f310 and an older version of the dual action at the same time. One or the other will work, but once I configure one, then the other won't recognize any buttons and Console shows errors like
 ```
12/6/17 7:14:08.222 PM OpenEmu[1772]: Controller type 'OEGenericDeviceIdentifier_1133_49686' does not recognize the event '<OEHIDEvent 0x7ff609793f80 pad=0x7ff609748fd0 type=Axis axis=Y direction=Neg value=-0.992188 'Y-' cookie=19>', when attempting to set the key with name: 'OESNESButtonUp'.
```
Same as for previous commenters, deleting the "OEControllerLogitechDualAction" dictionary from the Controller-Database.plist did not help

**Comment by clobber** [comment]
"Me toos" aren't adding anything to the discussion here besides an annoying email notification. We [are aware](https://github.com/OpenEmu/OpenEmu/issues/2326#issuecomment-240236674) there is a bug with this particular configuration, which is why the issue is still open. That means it is not fixed.

**Comment by clobber** [comment]
Officially fixed as of OpenEmu 2.2.
